seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java

```diff
@@ -79,7 +87,14 @@ public void write(@NonNull SeaTunnelRow seaTunnelRow) {
             } else {
                 fsDataOutputStream.write(rowDelimiter.getBytes());
             }
-            fsDataOutputStream.write(serializationSchema.serialize(seaTunnelRow));
+            SeaTunnelRow seaTunnelRowNew = new SeaTunnelRow(sinkColumnsIndexInRow.size());
```

Comment:
Here you can determine whether the number of SeaTunnelRow fields and the number of sinkColumnsIndexInRow are consistent, if consistent, can use the origin row not the new row. This will improve performance.

Reply:
done, good idea.